### PR TITLE
AB#83673 Use Accounts in AHP via http

### PR DESCRIPTION
**Pull request by jjankowski87**

### CODE REVIEW

File: src/HE.Investments.Account.WWW/Controllers/UserApiController.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+using HE.Investments.Common;
+using HE.Investments.Common.Contract;
+using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.FeatureManagement;
+
+namespace HE.Investments.Account.WWW.Controllers;
+
+[AllowAnonymous]
```
Comment (Antek012) [inline src/HE.Investments.Account.WWW/Controllers/UserApiController.cs:13]: Is ApiController attribute not needed?

**Review by Antek012 — APPROVED** [review]

File: src/HE.Investment.AHP.Domain/Programme/AhpProgrammeRepository.cs [hunk 2]
```diff
@@ -1,24 +1,34 @@
...
     public async Task<AhpProgramme> GetProgramme(AhpApplicationId applicationId, CancellationToken cancellationToken)
     {
-        var programme = await _programmeRepository.GetCurrentProgramme(ProgrammeType.Ahp, cancellationToken);
+        var programme = await _programmeService.Get(cancellationToken);
+        if (programme == null)
+        {
+            throw new InvalidOperationException("Cannot find Programme.");
+        }
+
+        if (programme.startOn == null || programme.endOn == null)
```
Comment (Carq) [inline src/HE.Investment.AHP.Domain/Programme/AhpProgrammeRepository.cs:26]: Shouldn't it be part of AhpProgramme?

File: src/HE.Investments.Account.Api.Contract/HE.Investments.Account.Api.Contract.csproj [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    <AnalysisMode>Default</AnalysisMode>
+    <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(Configuration)'!='DEBUG'">
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+  </PropertyGroup>
+
+  <ItemGroup>
```
Comment (Carq) [inline src/HE.Investments.Account.Api.Contract/HE.Investments.Account.Api.Contract.csproj:17]: Now we have "Directory.Build.props` file and part of those properties could be putted there.

**Review by Carq — APPROVED** [review]

**Review by dawid-pawulski — APPROVED** [review]

File: src/HE.Investment.AHP.Domain/Programme/AhpProgrammeRepository.cs (hunk 2 above)
Comment (jjankowski87) [reply src/HE.Investment.AHP.Domain/Programme/AhpProgrammeRepository.cs:26]: moved to entity

File: src/HE.Investments.Account.Api.Contract/HE.Investments.Account.Api.Contract.csproj (hunk 3 above)
Comment (jjankowski87) [reply src/HE.Investments.Account.Api.Contract/HE.Investments.Account.Api.Contract.csproj:17]: I'll do this in separate PR as it introduces a lot of changes.

**Pull request merged**